sdks/python/apache_beam/runners/sdf_utils.py

```diff
@@ -0,0 +1,173 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pytype: skip-file
+
+"""Common utility class to help SDK harness to execute an SDF. """
+
+from __future__ import absolute_import
+from __future__ import division
+
+import logging
+import threading
+from builtins import object
+from collections import namedtuple
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import Optional
+from typing import Tuple
+
+from apache_beam.utils import timestamp
+
+if TYPE_CHECKING:
+  from apache_beam.io.iobase import RestrictionTracker
+  from apache_beam.utils.timestamp import Timestamp
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SplitResultPrimary = namedtuple(
+    'SplitResultPrimary', 'windowed_value')
```

Comment:
why use a `namedtuple` here if there is only one member?

Reply:
To keep the similar structure with `SplitResultResidual`.

Reply:
I'll leave the final call on that up to @robertwb, but my instinct is to use a type alias here, unless there's some need to do tuple operations on this type.

I'm not sure of the exact type of windowed_value here, but a type alias would be something like:

```python
SplitResultPrimary = WindowedValue
```

Reply:
Nothing comes to mind immediately as to what fields we'd want to add here (though originally even the residual didn't have anything extra). Mostly I like the consistency, so I'd lean towards keeping it as is.